server/modules/authentication/discord/authentication.js

```diff
@@ -19,9 +22,16 @@ module.exports = {
         passReqToCallback: true
       }, async (req, accessToken, refreshToken, profile, cb) => {
         try {
-          if (conf.guildId && !_.some(profile.guilds, { id: conf.guildId })) {
+          if (conf.roles) {
+            const authRoles = conf.roles.split();
+            const { roles } = await discord.getGuildMember(accessToken, conf.guildId);
+            if (authRoles.every(role => roles.includes(role))
+              throw new WIKI.Error.AuthLoginFailed()
+            }
```

Comment:
I've got one more thing to report 😅 since the `if` has changed and it doesn't contain opening bracket this closing bracket should not be there.

```suggestion
```